unity-renderer/Assets/Scripts/MainScripts/DCL/Models/AvatarAssets/WearableItem.cs

```diff
@@ -231,6 +238,14 @@ public string[] GetHidesList(string bodyShapeType)
                 : hides.Concat(SKIN_IMPLICIT_CATEGORIES).Distinct().ToArray();
         }
 
+        // Old UPPER_BODY Wearables are incompatible with the new Hands category,
+        // so we apply this rule to hide the hands by default
+        bool isOrHidesUpperBody = hides.Contains(Categories.UPPER_BODY) || data.category == Categories.UPPER_BODY;
+        bool removesDefault = data.removesDefaultHiding?.Contains(Categories.HANDS) ?? false;
+
+        if (isOrHidesUpperBody && !removesDefault)
+            hides = hides.Concat(UPPER_BODY_DEFAULT_HIDES).Distinct().ToArray();
```

Comment:
we might be able to improve GC if we store the hides information in a pre-allocated buffer

Reply:
I didn't want to touch the original implementation but now that you mention it, all of this should be just a `hashset`